hardware/arduino/avr/libraries/HID/HID.cpp

```diff
@@ -130,24 +110,16 @@ bool HID_Setup(USBSetup& setup, uint8_t i)
 	}
 }
 
-HID_::HID_(void)
-{
-	static uint8_t endpointType[1];
-
-	endpointType[0] = EP_TYPE_INTERRUPT_IN;
-
-	static PUSBCallbacks cb = {
-		.setup = &HID_Setup,
-		.getInterface = &HID_GetInterface,
-		.getDescriptor = &HID_GetDescriptor,
-		.numEndpoints = 1,
-		.numInterfaces = 1,
-		.endpointType = endpointType,
-	};
-
-	static PUSBListNode node(&cb);
+// XXX: I've found no way to pass literal value directly in
+//      the PUSBListNode constructor
+static uint8_t epTypeDef[] = { EP_TYPE_INTERRUPT_IN };
 
-	HID_ENDPOINT_INT = PUSB_AddFunction(&node, &HID_INTERFACE);
+HID_::HID_(void) : PUSBListNode(1, 1, epTypeDef),
+                   rootNode(NULL), sizeof_hidReportDescriptor(0),
+                   modules_count(0), protocol(1), idle(1)
+{
+	// XXX: Shall this be done in PUSBListNode(...) constructor?
```

Comment:
No, this should not be done in the constructor. That's how I tried it before.

The problem is, that epTypeDef is a pointer/array. When you inherit the PUSBListNode and call the plug() function at the same moment it initializes the endpoint with this array value. But if the array is not initialized before, it just ignores its content. Because of this reason we need to manually call the add function ourselves.

So the problem is to initialize an array inside a constructor.
There was another possible workaround with a static array length member etc but I think this all makes it more complicated.

If we miss this feature and only accept one endpoint this could reduce the flexibility (I think Midi needs 1 Interface + 2 endpoints).

My suggestion:
Add an array with size 1 to the HID class and initzialize it inside the constructor function, before the plug() function. That should work.